File: Easy/125_ValidPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        List<Character> list = new ArrayList<>();

        for (Character ch : s.toCharArray()) {
            if (Character.isLetterOrDigit(ch)) {
                list.add(Character.toLowerCase(ch));
            }
        }

        if (list.size() < 2) {
            return true;
        }

        for (int i = 0; i < list.size() / 2; i++) {
            if (list.get(i) != list.get(list.size() - 1 - i)) {
                return false;
            }
        }

        return true;
    }
}

File: NonJava/125_ValidPalindrome.ts
function isPalindrome(s: string): boolean {
    if (s.trim().length === 0) {
        return true;
    }

    const array: Array<string> = [];

    for (let i = 0; i < s.length; i++) {
        if (s.charAt(i).match(/[a-zA-Z0-9]/)) {
            array.push(s.charAt(i).toLowerCase());
        }
    }

    for (let i = 0; i < array.length / 2; i++) {
        if (array[i] !== array[array.length - 1 - i]) {
            return false;
        }
    }

    return true;
};